tests/runtime/test_subbuffers.cpp

```diff
@@ -145,10 +154,7 @@ int TestOutputDataDecomposition() {
 
     std::vector<int> AfterSubBufCContents(NumData);
 
-    for (size_t i = 0; i < Queues.size(); ++i)
-      Queues[i].finish();
-
-    Queues[0].enqueueReadBuffer(CBuffer, CL_TRUE, 0, sizeof(cl_int) * NumData,
+    Queues[0].enqueueReadBuffer(CBuffer, CL_FALSE, 0, sizeof(cl_int) * NumData,
```

Comment:
Seems this explicitly does *not* synchronize between queues since they use separate non-overlapping sub-buffers? It's probably a good idea to explain this at the start of the file, since it's likely not something most people would expect in an OpenCL application.